app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -83,18 +82,18 @@ def not_found(options = nil)
   end
 
   def registration_url
-    uri = if params[:url].present?
-            URI.join(params[:url], '/register')
-          else
-            URI(register_url)
-          end
+    uri = URI(params[:url] || root_url)
 
-    return uri if uri.scheme && uri.host
+    return URI.join(uri, 'register') if uri.scheme && uri.host
 
     msg = N_('URL in :url parameter is missing a scheme, please set http:// or https://')
     fail Foreman::Exception.new(msg)
   end
 
+  def context_urls
+    { server_url: URI(root_url), registration_url: registration_url }
```

Comment:
server is a highly ambiguous word in the Foreman context. Can we be more explicit which server? I also think it's no longer used now?